types/ndarray/index.d.ts

```diff
@@ -6,60 +6,67 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-declare function ndarray<T = number>(
-    data: ndarray.Data<T>,
+declare function ndarray<D extends ndarray.Data = ndarray.Data<number>>(
+    data: D,
     shape?: number[],
     stride?: number[],
     offset?: number,
-): ndarray.NdArray<T>;
+): ndarray.NdArray<D>;
 
 declare namespace ndarray {
-    interface NdArray<T = number> {
-        data: Data<T>;
+    interface NdArray<D extends Data = Data<number>> {
```

Comment:
The default type `Data<number>` ensures backwards compatibility when using `NdArray` as an argument or return type without a generic.